Adding initial 508 Compliant MCP statements page

Dr-Pongo opened the pull request:
## Description
Creating new page with navigation for HTML Statements page

## Original issue(s)
department-of-veterans-affairs/va.gov-team#35871

## Acceptance criteria
- [ ] Bare bones page and navigation is implemented

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

Dr-Pongo committed:
Added page  still WIP

Dr-Pongo committed:
Added page  still WIP

Dr-Pongo committed:
Added page  still WIP

Dr-Pongo committed:
Functional bare bones page

Dr-Pongo committed:
Functional bare bones page

Dr-Pongo committed:
Functional bare bones page

Dr-Pongo committed:
Functional bare bones page

mdewey commented on src/applications/medical-copays/components/HTMLStatementList.jsx:
```diff
@@ -0,0 +1,55 @@
+
+const HTMLStatementList = () => {
+  const { pathname } = useLocation();
+  const selectedId = pathname.replace('/balance-details/', '');
```
There must be a better way to get the URL param.

Dr-Pongo replied:
Passing in ID instead of pulling from URL since that's all this was doing

mdewey commented on src/applications/medical-copays/components/HTMLStatementList.jsx:
```diff
@@ -0,0 +1,55 @@
+  const selectedId = pathname.replace('/balance-details/', '');
+  const statements = useSelector(({ mcp }) => mcp.statements) ?? [];
+  // get selected statement
+  const [selectedCopay] = statements?.filter(({ id }) => id === selectedId);
```
Do we know when statements would be `falsey`?  I understand why the Elvis (`?.`) is here, but ifs null here it will crash down the page. 

If statements can be nothing, we should plan accordingly

Dr-Pongo replied:
Nuked the Elivs checks, you wouldn't be able to make it to this page if statements were falsey

mdewey commented on src/applications/medical-copays/components/HTMLStatementList.jsx:
```diff
@@ -0,0 +1,55 @@
+  // get facility  number on selected statement
+  const facilityNumber = selectedCopay?.pSFacilityNum;
+  // filter out all statements that are not related to this facility
+  const facilityCopays = statements?.filter(
```
The filter and sorts are good places for test (unit and/or cypress

Dr-Pongo replied:
Going to write up a ticket to make unit tests for these! Thank you!

mdewey commented on src/applications/medical-copays/components/HTMLStatementList.jsx:
```diff
@@ -0,0 +1,55 @@
+            data-testid="statement"
+            key={statement.id}
+          >
+            <span aria-hidden="true">
```
This could probably be its own component

Dr-Pongo replied:
Busted the link out to a separate component in the same file in [538093a](https://github.com/department-of-veterans-affairs/vets-website/pull/20484/commits/538093aa97485ccd50b6b2b4dcd2f73b0cb9048d)

mdewey reviewed (changes requested):
Somethings to clean up but not bad! Good work!

AdamFreemer committed:
Re-order filters for gi tool (#20487)

Co-authored-by: Adam Freemer <adam.freemer@accenturefederal.com>

Mottie committed:
NOD: Fix submitted profile data length (#20489)

* Truncate profile email, phone and address before submission

* Add & update unit tests

andresriveratoro committed:
chore: update component-library dep to 7.4.0 (#20483)

andresriveratoro committed:
chore: remove hideCloseButton node (#20486)

theodur committed:
Enable changed app build deploys (vagovdev) (#20479)

* Add partial deploy script

* Update manual dev deploy for testing

* Test manual dev deploy with single app build

* Don't notify failure for testing

* Update indentation

* Fix argument check

* Remove extra flag

* Exit successfully

* Sync to rev proxy bucket before compressing

* Add script for removing global assets in archive

* Make filtered directory

* Update directory

* Update paths in rsync

* Update remove assets

* ls directory

* use bash instead of shell

* Refactor and clean up

* Update env variables in GHA

* Test with another app

* Update message

* Update CI workflow to use remove global asset script

* Test with single app build

* Filter paths in cypress test selection for testing

* Update logic

* Add checkout step to archive job

* Test multiple app builds

* Verify archive job will have outputs on master

* Comment testing reports cypress

* Revert "Comment testing reports cypress"

This reverts commit 5451d7e9f4def5868311eac7fd9122a15baec2ca.

* Revert "Verify archive job will have outputs on master"

This reverts commit 099d0110437febfaca2f1a6f6f77f01161ff34d2.

* Update comments

* Revert testing changes

* Update deploy job to use partial-deploy script

dsrr47 committed:
33866  Links and labels should be sentence cased (#20485)

* text updates

* updated button role

* added keydown handler

Co-authored-by: David <dsrr74@gmail.com>

theodur committed:
Update logic in Deploy job (#20493)

* Update if statement

* Small update

* Revert "Small update"

This reverts commit 1af6a48bff1c7b2aef42c352c93752733db260c4.

cohnjesse committed:
Adjusted uploader to send files and show success message (#20376)

* Adjusted uploader to send files and show success message

* updated unit test

* Fixed spelling

* Changed to send string

* Changed to file

* sending single file

* sending encoded data

* changed content type

* changed again

* json stringified

* Added document type

* Added more params

* refactored a bit

* Added ability for multiple files to be submitted at once

* Removed console logs and added error messaging

* updated state and error message

* refactored

* Removed redux

* Added e2e tests for document uploader

* Removed duplicate from e2e test

* Updated unit test

* Removed unit tests moving to e2e tests

* Skipping tests till we are in production

* Skipping tests till we are in production

* Skipping unit we are in production

* removed commented code

* Made comment easier to read

* Changed alert to match mockups

* refactored

* Updated success text

theodur committed:
Add quotes to variable (#20496)

theodur committed:
Update operator (#20497)

jhonny218 committed:
Unused CSS Removal v4 (#20488)

tlei123 committed:
Integrate My-VA Appointments E2E-specs with TestRail. (#20495)

theodur committed:
Updates list changed files steps (#20501)

SebVortex committed:
24662 10203 direct deposit (#20500)

* Changed bottom accordion text

* Pre submit info WIP

* Forgot React import

* custom presubmit info object used for conditional change

dsrr47 committed:
33866 sentence case (#20499)

* text updates

* updated button role

* added keydown handler

* changed text to be cased correctly

Co-authored-by: David <dsrr74@gmail.com>

SebVortex committed:
Removed env flags for 23485 (#20503)

justin-trieu committed:
[department-of-veterans-affairs/va-virtual-agent#412] Changes google tag for submit and speak with an agent buttons (#20505)

Co-authored-by: Jacob Gacek <jacob.gacek@thoughtworks.com>

Co-authored-by: Jacob Gacek <jacob.gacek@thoughtworks.com>

Mottie committed:
HLR/NOD: Replace typographical quote (#20507)

* Replace typographical apostrophe with single quote

* Update unit tests & mock data

allisonlu committed:
VACMS-7878: Offices - Search filter input (#20491)

* VACMS-7878:
- created a /components/ directory separate from containers
- moved ChildrenOfficeList

* VACMS-7878
- created an OfficeList component separate from App laucnhing point
- memoized office list for performance

* VACMS-7878:
- created a SearchForm (behaves more like a filter, will refactor later)
- displays list of all offices
- has input field for users to find desired office

* VACMS-7878
- cleaned up App.jsx

* VACMS-7878: clean up

* VACMS-7878: added Back to Top component

* VACMS-7878: changed component name of SearchForm to SearchFilter, and some variable names as well for better readability

* VACMS-7878
- replaced a h1 with paragraph
- removed extra code

* VACMS-7878: added id to pair with htmlFor label attribute

justin-trieu committed:
Virtual agent 412 tagging (#20511)

cesaresantiago committed:
Send default date in day format. (#20510)

Current code sends a default date in non string format for a timeout, which crashes the app. We need to send it in "year-month-day" string format.

theodur committed:
Add workflow for uploading cross app import report (#20498)

ddzz committed:
Upgrade actions for uploading and downloading CI artifacts to latest version (#20509)

mdewey committed:
(feature) added phone error message (#20467)

* (feature) added phone error message

* (test) updated branch for API testing

* (chore) trigger build process

* (chore) trigger build process 2

* (test) revert jenkfile to master

* (feature) added UUID call to API call

* (chore) cleaned old file

* (test) added test for update button

* (test) added test for app constants

* (featurte) added logic for changing heade

* (test) updated test for phone numbe

* (feature) adjusted required logic

* (feature) updated what edit button looks like

* (feature) updated email header

ddzz committed:
Use ESBuild for Cypress tests (#20420)

Dr-Pongo committed:
Broke out link component, cleaned up statement ID retrival

Dr-Pongo committed:
Broke out link component, cleaned up statement ID retrival

Dr-Pongo committed:
Broke out link component, cleaned up statement ID retrival

mdewey reviewed (commented):
Not bad, but missing tests. I would say just to start you add axeCheck to your components (https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/applications/check-in/components/pages/Edit/tests/Address.Edit.unit.spec.jsx#L14) and at least a e2e test to test the feature flip

ddzz committed:
Add data validation to the Cypress form tester (#20469)

dsrr47 committed:
removed env flags (#20502)

Co-authored-by: David <dsrr74@gmail.com>

jerekshoe committed:
Adding privacy policy to Review page (#20506)

SebVortex committed:
Changed routing number title and fixed note formatting (#20513)

Also added dash to 0994 between 9 and digit after noticing it wasn't there.

Commit:
33774 address change messaging help text and modal (#20481)

* add hint text to fields

* update modal content for address change

* mock success and failure modal UI components

* adjust field text and hint text

* remove logging

* update modal styling to match design

OptionSelect committed:
Add VA-On-ThisPage for 35873

OptionSelect committed:
Add VA-On-ThisPage for 35873

tlei123 committed:
Integrate My-VA Benefits E2E-spec with TestRail. (#20508)

OptionSelect committed:
Fixed conflicts

asg5704 committed:
Adds unit tests for LoginHeader (#20463)

* initial commit

* adds LoginHeader unit tests

* removes wrapper.unmount for last unit test

FernandoBoza committed:
Feature/education inbox letters UI (#20229)

* Created Layout component, InboxPage container,

* Imported and refactored App with Layout component and updated routes.jsx with Switch,Route

* fleshing out download edu inbox letters UI

* wrap InboxPage.jsx around in Redux connect, with letters action. Fleshing out the res

* changed method name to FETCH_CLAIM_STATUS

* added claim endpoint to download letters anchor

* updated breadcrumbs component

* fixes btn with className and added conditional UI

* updated app with loading spinner and letters download endpoint

* brought in moment to format received_date from api.

* added visitor/login redirect validation logic.

* removed moment and used date-fn instead

Co-authored-by: fer.boza <fernando.boza@accenturefederal.com>

Kwoly committed:
[DON'T MERGE] 37930: Make Login.gov visible by default and hide via flipper (#20452)

* make login gov visible by default

* add login.gov to string test case

Mottie committed:
526: Update to use `va-telephone` (#20517)

* Update to va-telephone & fix lint issues

* Fix unit test

annaswims committed:
Add canceled to list of intent to file statuses (#18929)

Co-authored-by: Robin Garrison <robin.garrison@adhocteam.us>

rjohnson2011 committed:
Instrument Code & Add Cypress Code Coverage Report to Github Actions/CI Workflow (#20504)

* Babelrc and cypress mods for code coverage

* Required dev dependency packages

* Update yarn

* CI passes no errors, no out json not found

* Create coverage artifact

* Always upload coverage artifact

* Instanbul in dev dependencies

* Upload coverage summary not entire folder artifact

* Move istanbul to babelrc

* PR refactors

* Fix syntax

* Rerun yarn

* Linting

* Linting fixes

timwright12 committed:
Updated CRON for daylight savings time (#20522)

ddzz committed:
Remove `@cypress/webpack-preprocessor` (#20523)

longmd committed:
Add Login Widget for Home Loans COE (#20518)

tmitchellgcio committed:
38167 healthcare notification and messages (#20520)

* Update Url

* appointments and notifications as well as tests

* fix tests

Commit:
rename: changed the name of the formulate repo and npm package name (#20524)

* rename: changed the name of the formulate repo and npm package name

* removing entry in yarn.lock and replacing with correct version of npm package that was renamed

tlei123 committed:
Add Node utility script to count Cypress E2E specs (#20519)

* Add count-cy-specs Node-script.

* Refactor count-cy-specs util-script variables.

* Update count-cy-specs util-script to find form-tester CI-only skips.

* Add progress-spinner to count-sy-specs util-script.

* Add count-cy-specs util-script call to package.json.

* Add count-cy-specs util-script call info to README.

mdewey committed:
(test) added test replicate and verify bug (#20514)

Co-authored-by: Adrian Rollett <acrollet@gmail.com>

acrollet committed:
Add translation feature flags (#20515)

Dr-Pongo committed:
Broke out Statement link, added Cypress tests

OptionSelect committed:
Add statement addresses

OptionSelect committed:
Add statement addresses

Dr-Pongo committed:
Merge remote-tracking branch 'origin/35871-html-statements' into 35871-html-statements

Dr-Pongo commented:
Closing to clean up failed rebase!
